packages/hoppscotch-common/src/components/collections/Properties.vue

```diff
@@ -163,4 +171,41 @@ const saveEditedCollection = () => {
 const hideModal = () => {
   emit("hide-modal")
 }
+
+const onGenerateOAuthToken = () => {
+  const localOAuthTempConfig =
+    persistenceService.getLocalConfig("oauth_temp_config")
+
+  if (!localOAuthTempConfig || !props.editingProperties) {
+    return
+  }
+
+  const persistedOAuthConfig: PersistedOAuthConfig =
+    JSON.parse(localOAuthTempConfig)
+
+  // Only Grant Types involving redirects require persisting associated information
+  if (
+    persistedOAuthConfig.grant_type &&
+    !grantTypesInvolvingRedirect.includes(persistedOAuthConfig.grant_type)
+  ) {
+    return
+  }
+
+  // Write the collection object to `localStorage` to retrieve later while redirecting back post the OAuth flow
+  if (persistedOAuthConfig.context?.type === "collection-properties") {
+    persistenceService.setLocalConfig(
+      "oauth_temp_config",
+      JSON.stringify(<PersistedOAuthConfig>{
+        ...persistedOAuthConfig,
+        context: {
+          ...persistedOAuthConfig.context,
+          metadata: {
+            collection: props.editingProperties.collection,
+            collectionID: props.editingProperties.path,
```

Comment:
We could obtain the collection entry while getting redirected back after the OAuth flow from the ID (path) alone, but might require it entirely for teams; thought to keep it in the same format across.